docs/api/effector/guard.md

```diff
@@ -65,3 +61,53 @@ guard({
 submitForm('') // nothing happens
 submitForm('alice') // ~> searchUser('alice')
 ```
+
+# `guard({sourceEvent, {filter: filterStore})`
+
+#### Arguments
+
+1. `sourceEvent` (_Event_): Source event
+1. `filterStore` (_Store_): Filter store
+
+#### Example
+
+```js try
+const trigger = createEvent()
+const $unlocked = createStore(true)
+const {lock, unlock} = createApi($unlocked, {
+  lock: () => false,
+  unlock: () => true,
+})
+
+const target = guard(trigger, {
+  filter: $unlocked,
+})
+
+target.watch(console.log)
+trigger('A')
+lock()
+trigger('B') // nothing happens
+unlock()
+trigger('C')
+```
+
+# `guard({sourceEvent, {filter: predicate})`
+
+#### Arguments
+
+1. `sourceEvent` (_Event_): Source eventre
+2. `filter` (([a]) => Boolean): Optional predicate function, should be **pure**
```

Comment:
filter is never optional
and better declare function like this:
`(sourcePayload) => Boolean`